atc/worker/worker.go

```diff
@@ -207,6 +212,68 @@ func (worker *gardenWorker) LookupVolume(logger lager.Logger, handle string) (Vo
 	return worker.volumeClient.LookupVolume(logger, handle)
 }
 
+func (worker *gardenWorker) imagePolicyCheck(
+	ctx context.Context,
+	delegate ImageFetchingDelegate,
+	metadata db.ContainerMetadata,
+	containerSpec ContainerSpec,
+	resourceTypes atc.VersionedResourceTypes,
+) (bool, error) {
+	if worker.policyChecker == nil {
+		return true, nil
+	}
+
+	// Actions in skip list will not go through policy check.
+	if worker.policyChecker.ShouldSkipAction(policy.ActionUsingImage) {
+		return true, nil
+	}
+
+	imageSpec := containerSpec.ImageSpec
+
+	imageInfo := map[string]interface{}{
+		"privileged": imageSpec.Privileged,
+	}
+
+	if imageSpec.ImageResource != nil {
+		imageInfo["image_source_type"] = imageSpec.ImageResource.Type
```

Comment:
nit: I find it confusing to say 'source type' when they're sibling fields.

```suggestion
		imageInfo["image_type"] = imageSpec.ImageResource.Type
```

Reply:
Renamed.